src/VSCodeExtension/src/formatter/format-document.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as vscode from "vscode";
+import { FormatRule, formatter } from "./formatter";
+import indentRules from "./rules/indent";
+import operationRules from "./rules/operation";
+import controlStructureRules from "./rules/control-structure";
+
+const rules: FormatRule[] = [
+    ...indentRules,
+    ...operationRules,
+    ...controlStructureRules,
+];
+
+export const formatDocument = (
+    document: vscode.TextDocument
+): vscode.TextEdit[] => {
+    const lines: vscode.TextLine[] = Array.from(
+        Array(document.lineCount),
+        (_, lineNo) => document.lineAt(lineNo)
+    );
+
+    const formattedLines: vscode.TextEdit[] = lines
+        .map((line) => {
+            const formattedLine: string = formatter(line.text, rules);
+            return formattedLine !== line.text
```

Comment:
Does this constrain all formatting rules to work one line at a time?

Reply:
Yup. I'm not sure how to match the whole document at once. Will definitely have to change this to handle multi-line matching.

Reply:
we could just go lines.join(""); counldn't we?

Reply:
It would be a bit more involved than that. We need to return `vscode.TextEdit[]` instead of just the final formatted string.